Src/GoogleApis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -99,5 +110,45 @@ public Initializer(string tokenUrl)
         }
 
         #endregion
+
+        /// <summary>
+        /// Return whether code is running on Google Compute Engine.
+        /// </summary>
+        public static async Task<bool> IsRunningOnComputeEngine()
+        {
+            try
+            {
+                Logger.Info("Checking connectivity to ComputeEngine metadata server.");
+                var httpRequest = new HttpRequestMessage(HttpMethod.Get, MetadataServerUrl);
+
+                CancellationTokenSource cts = new CancellationTokenSource();
+                cts.CancelAfter(MetadataServerPingTimeoutInMilliseconds);
+
+                var httpClient = new HttpClientFactory().CreateHttpClient(new CreateHttpClientArgs());

```

Comment:
By the way by using the HttpClientFactory, you already getting retry mechanism for free. I'm not sure that that's your intent. I think you might just want to use a regular HttpClient (without our wrapper), but I'm still not sure... what is this request exactly? I would add a note in the method comment that there is actually a request for x since the name itself doesn't imply for a http request at all!